src/Mvc/Mvc.Core/src/ControllerBase.cs

```diff
@@ -1821,6 +1822,52 @@ public virtual ConflictObjectResult Conflict([ActionResultObjectValue] object er
         public virtual ConflictObjectResult Conflict([ActionResultObjectValue] ModelStateDictionary modelState)
             => new ConflictObjectResult(modelState);
 
+        /// <summary>
+        /// Creates an <see cref="ObjectResult"/> that produces a <see cref="ProblemDetails"/> response with a <c>500</c>
+        /// error status with a <see cref="ProblemDetails" /> value.
+        /// </summary>
+        /// <param name="title">The value for <see cref="ProblemDetails.Title" />.</param>
+        /// <param name="type">The value for <see cref="ProblemDetails.Type" />.</param>
+        /// <param name="detail">The value for <see cref="ProblemDetails.Detail" />.</param>
+        /// <param name="instance">The value for <see cref="ProblemDetails.Instance" />.</param>
+        /// <returns>The created <see cref="ObjectResult"/> for the response.</returns>
+        [NonAction]
+        public virtual ObjectResult Problem(
+            string detail = null,
+            string instance = null,
+            string title = null,
+            string type = null)
```

Comment:
does it make sense to make the status code a parameter too?

Users can set the status code on the object result, but that won't apply the defaults.